public/mobile-app/src/lib/agenda.ts

```diff
@@ -0,0 +1,209 @@
+    )
+  }
+
+  get title(): String {
+    return this._title
+  }
+
+  get description(): String {
+    return this._description
+  }
+
+  get date(): Date {
+    return this._start_date || this._date
+  }
+
+  get dayName(): String {
+    return this.date.toLocaleDateString('fr-FR', { weekday: 'short' }).replace('.', '')
+  }
+
+  get dayNum(): String {
+    return this.date.getDate()
+  }
+
+  get monthName(): String {
+    return monthName(this.date)
+  }
+
+  get period(): String {
+    const locale = 'fr-FR'
+    let startFormat = { year: 'numeric', month: 'long', day: 'numeric' }
+    let dateFormat = startFormat
+    if (this._start_date) {
+      const start = this._start_date.toLocaleDateString(locale, startFormat)
+      if (this._end_date) {
+        let endFormat = startFormat
+        if (this._start_date.getYear() == this._end_date.getYear()) {
+          startFormat = { month: 'long', day: 'numeric' }
+          if (this._start_date.getMonth() == this._end_date.getMonth()) {
+            startFormat = { day: 'numeric' }
+          }
+        }
+        const start = this._start_date.toLocaleDateString(locale, startFormat)
+        const end = this._end_date.toLocaleDateString(locale, endFormat)
+        return `Du ${start} au ${end}`
+      }
+      return `À partir du ${start}`
+    }
+    const date = this._date.toLocaleDateString(locale, dateFormat)
+    return date
+  }
+
+  private static readonly MAP: Record<Kind, { label: string; icon: string }> = {
+    holiday: {
+      label: 'Vacances et jours fériés',
+      icon: 'fr-icon-calendar-event-fill',
+    },
+    otv: {
+      label: 'Logement',
+      icon: 'fr-icon-home-4-fill',
+    },
+  }
+
+  get kind(): Kind {
+    return this._kind
+  }
+
+  get label(): String {
+    const map = Item.MAP[this._kind]
+    if (map === undefined) {
+      return ''
+    }
+    return map.label
+  }
+
+  get icon(): String {
+    const map = Item.MAP[this._kind]
+    if (map === undefined) {
+      return ''
+    }
+    return map.icon
+  }
+}
+
+export class Agenda {
+  private _now: Item[] = []
+  private _next: Item[] = []
+
+  constructor(holidays: Holiday[], date: Date | null = null) {
+    let today = date || new Date()
+    today.setHours(0, 0, 0, 0)
+    const oneday_in_ms = 24 * 60 * 60 * 1000
+    let items: Item[] = []
+
+    // build items from holidays
+    holidays.forEach((holiday) => {
+      // convert dates
+      let title = holiday.description
+      if (holiday.zones) {
+        title += ' ' + holiday.zones
+      }
+      if (holiday.emoji) {
+        title += ' ' + holiday.emoji
+      }
+      let item = new Item(
+        'holiday',
+        title,
+        null,
+        null,
+        holiday.start_date,
+        holiday.end_date
+      )
+      items.push(item)
+    })
+
+    // create OTV items
+    let seenHolidays = new Set()
+    holidays.forEach((holiday) => {
+      let key = JSON.stringify({ desc: holiday.description, date: holiday.date })
```

Comment:
This is weird, didn't you set the holiday's description and date to `null` in the loop just above?
How does it work, shouldn't it just add a single OTV instead of one per holiday? I don't understand this.